### ISSUE TITLE
cmake doesn't find gettext (msgmerge) version due to .exe extension

### ISSUE THREAD

**Issue by KrullBorg** [Alexpux/MINGW-packages#2435]
The cmake module FindGettext cannot find the gettext (msgmerge) version due to .exe extension, because it search it without extension (https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindGettext.cmake#L69).
What could be the right thing to do? patch cmake module or create symbolic links to msgmerge.exe without .exe?
I ask it because i didn't know cmake

**Comment by peterbud** [comment]
Hi @KrullKorg . For darktable buid it works (uses `cmake`, but tries to find `msgfmt`).
Here is the appropriate [cmake code](https://github.com/darktable-org/darktable/blob/6ea20f75a265442c5649415b1a46bc75a96c2da8/CMakeLists.txt#L421)
And here you can see the [build output](https://ci.appveyor.com/project/peterbud/darktable/build/1.0.133#L579)

**Comment by KrullKorg** [comment]
it isn't the same thing: the package that i want to build (evolution-data-server) uses the cmake module FindGettext; and as you can see from the module's source it searches msgmerge without extension; with your solution i must patch each package that use this module; i think is better to change cmake module or gettext executables name (in this case simply with a symbolic link, so other packages continue to work)

**Comment by peterbud** [comment]
Have you tried to change the `FindGettext() `to `find_package(Gettext)`? If that works, that would be much easier IMHO.

**Comment by KrullKorg** [comment]
i'll try, but the "problem" persists: i must patch each package that use findgettext

**Comment by Ede123** [comment]
I'd say there are two (three) sensible solutions:
* If the change is useful for all projects using cmake, then a fix should be created and submitted upstream to achieve consistent behavior.
* If it's only useful for some projects, then patch those projects and be done with it.
* (patch gettext itself - the fact that the output of `msgmerge --version` changes across systems and is not consistent makes it obviously hard to parse and is therefore not ideal to start with)

Personally I'm not opposed to the first option as I encountered the very same issue some time ago (it was a project-specific copy of the module, but annoying nonetheless...

**Comment by lazka** [comment]
From what I see this was fixed upstream: https://gitlab.kitware.com/cmake/cmake/merge_requests/1192/diffs

And the current version in the pacman repo includes this fix.
